packages/agents-hosting/test/hosting/oauthflow.test.ts

```diff
@@ -0,0 +1,190 @@
+import { strict as assert } from 'assert'
+import { describe, it, beforeEach, afterEach } from 'node:test'
+import { CloudAdapter, FlowState, MemoryStorage, OAuthFlow, SigningResource, TokenRequestStatus, TurnContext, UserState, UserTokenClient } from './../../src'
+import { Activity, ActivityTypes } from '@microsoft/agents-activity'
+import sinon from 'sinon'
+
+const createTestActivity = () => Activity.fromObject({
+  type: ActivityTypes.Message,
+  channelId: 'test',
+  recipient: {
+    id: 'testRecipient'
+  },
+  serviceUrl: 'https://test.com',
+  from: {
+    id: 'testUser'
+  },
+  conversation: {
+    id: 'testConversation'
+  },
+  replyToId: 'testReplyToId',
+  text: 'testText'
+})
+
+const testSigninResource : SigningResource = {
+  signInLink: 'https://test.com',
+  tokenExchangeResource: {
+    id: 'testTokenExchangeId',
+    uri: 'https://test.com',
+  },
+  tokenPostResource: {
+    sasUrl: 'https://test.com',
+  }
+}
+
+describe('OAuthFlow', () => {
+  const userState = new UserState(new MemoryStorage())
```

Comment:
Ideally I think this should be mocked as well. You can mark it as pending and revisit later, I wouldn't block on it